app/models/static_content.rb

```diff
@@ -5,6 +5,24 @@ class StaticContent < ApplicationRecord
   validates :name, uniqueness: { case_sensitive: false }
 
   scope :filter_by_type, ->(type) { where data_type: type }
+  scope :ordered_by_name, ->(order) { order(name: order) }
+  scope :ordered_by_id, ->(order) { order(id: order) }
+
+  def self.for_params(params)
+    static_contents = StaticContent.all
+
+    if params[:type].present?
+      static_contents = static_contents.filter_by_type(params[:type])
+    end
+
+    if params[:id].present?
+      static_contents = static_contents.ordered_by_id(params[:id].to_sym)
+    elsif params[:name].present?
+      static_contents = static_contents.ordered_by_name(params[:name].to_sym)
+    end
+
+    static_contents
+  end
```

Comment:
The attribute `:id` is somewhat preloaded in REST and rails, so I would rather use something like `:sort_XXX` for the params.

e.g.: `:sort_column [id|name]`  and `:sort_order [asc|desc]`

This way you are also a bit more flexible to eventually add more sorting columns later on.

And by adding a whitelist of all possible sort_column names, it should be secure enough for an SQL injection.

Reply:
Resolved in #271